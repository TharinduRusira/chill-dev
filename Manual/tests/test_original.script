source: mm.c
procedure: mm
format: rose
loop: 0

known(ambn > 4)
known(an > 0)
known(bm > 0)
original()
print
