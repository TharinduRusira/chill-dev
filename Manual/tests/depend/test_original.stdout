for(t2 = 0; t2 <= an-1; t2++) {
  for(t4 = 0; t4 <= bm-1; t4++) {
    s2(t2,t4,0);
    s3(t2,t4,0);
    s4(t2,t4,1);
    s5(t2,t4,2);
    s6(t2,t4,3);
    for(t6 = 4; t6 <= ambn-1; t6++) {
      s1(t2,t4,t6);
    }
  }
}


for(t2 = 0; t2 <= an-1; t2++) {
  for(t4 = 0; t4 <= bm-1; t4++) {
    s2(t2,t4,0);
    s3(t2,t4,0);
    s4(t2,t4,1);
    s5(t2,t4,2);
    s6(t2,t4,3);
    for(t6 = 4; t6 <= ambn-1; t6++) {
      s1(t2,t4,t6);
    }
  }
}


dependence graph:
s0->s1: C:true(0, 0, +) C:true(0, 0, 0) C:output(0, 0, +) C:output(0, 0, 0)
s1->s1: C:anti(0, 0, +) C:output(0, 0, +)
s2->s1: C:true(0, 0, *) C:true(0, 0, *) C:output(0, 0, *) C:output(0, 0, *)
s2->s3: C:true(0, 0, +) C:true(0, 0, 0) C:output(0, 0, +) C:output(0, 0, 0)
s2->s4: C:true(0, 0, *) C:true(0, 0, *) C:output(0, 0, *) C:output(0, 0, *)
s2->s5: C:true(0, 0, *) C:true(0, 0, *) C:output(0, 0, *) C:output(0, 0, *)
s2->s6: C:true(0, 0, *) C:true(0, 0, *) C:output(0, 0, *) C:output(0, 0, *)
s3->s1: C:anti(0, 0, *) C:output(0, 0, *)
s3->s3: C:anti(0, 0, +) C:output(0, 0, +)
s3->s4: C:anti(0, 0, *) C:output(0, 0, *)
s3->s5: C:anti(0, 0, *) C:output(0, 0, *)
s3->s6: C:anti(0, 0, *) C:output(0, 0, *)
s4->s1: C:anti(0, 0, *) C:output(0, 0, *)
s4->s4: C:anti(0, 0, +) C:output(0, 0, +)
s4->s5: C:anti(0, 0, *) C:output(0, 0, *)
s4->s6: C:anti(0, 0, *) C:output(0, 0, *)
s5->s1: C:anti(0, 0, *) C:output(0, 0, *)
s5->s5: C:anti(0, 0, +) C:output(0, 0, +)
s5->s6: C:anti(0, 0, *) C:output(0, 0, *)
s6->s1: C:anti(0, 0, *) C:output(0, 0, *)
s6->s6: C:anti(0, 0, +) C:output(0, 0, +)
